pkg/trace/config/testdata/full.yaml

```
@@ -26,6 +26,9 @@ apm_config:
       - "apikey4              "
       - "apikey5\n \n         "
   env: test
+  otlp:
+    http_port: 50051
+    grpc_port: 50052
```

Is it worth making more explicit here that this is experimental?
```suggestion
  experimental:
    otlp:
      http_port: 50051
      grpc_port: 50052
```

Also, revisiting this, if we want to add more features that are HTTP or GRPC specific it will be easier if it's
```
http:
  port: 50051
```
instead of `http_port`

---

I'll address both your points:
1. The problem is that there's no going back from that config once we add it (`experimental`). If we really want to emphasise that, I'd rather add an extra layer on top such as feature flag.
2. On your second point. I'm on the fence. I tend to try and keep things simple. I can't think of something we might need, and maybe we'll never need anything, so I don't want to pre-maturely prepare for something that might not happen. Worst case scenario, we can always add the `http` group in parallel with `http_port` as a section for configuring HTTP behaviour and it wouldn't be that bad. I see it a little worse to add extra indentation and sections prematurely.

Open to discuss this further if you feel otherwise.

---

> I'd rather add an extra layer on top such as feature flag. We can also mention this in documentation
Both sound like good ideas to me

> I can't think of something we might need, and maybe we'll never need anything, so I don't want to pre-maturely prepare for something that might not happen

It may help to have a look at the OTLP receiver settings on the OpenTelemetry Collector https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/config.md we may not want to support most of those settings ever but if it's possible that we will want to support some of them in the future I think it's a good argument to use `http`/`grpc` groups

---

I've moved these under experimental, and left them like this. We can always change them now that we've concluded that experimental is bound to change and the user is aware of that.